src/DOMNodeComparator.php

```diff
@@ -119,10 +119,13 @@ public function assertEquals($expected, $actual, $delta = 0.0, $canonicalize = f
     private function nodeToText(DOMNode $node, $canonicalize, $ignoreCase)
     {
         if ($canonicalize) {
-            $document = new DOMDocument;
-            $document->loadXML($node->C14N());
+            $canonicalizedNode = (string) $node->C14N();
+            if ($canonicalizedNode !== "") {
```

Comment:
Probably not. The current check just seems saner te me, also in light of possible future changes, given that problems arise only when calling `->loadXML()` with an empty string. The cast is only necessary for HHVM, where `$emptyNode->C14N()` returns `null` instead of `""` for some reason. An alternative could be
```php
$canonicalizedNode = $node->C14N();
if ($canonicalizedNode !== "" && $canonicalizedNode !== null) {
```